art/attacks/evasion/composite_adversarial_attack.py

```diff
@@ -0,0 +1,669 @@
+# MIT License
+#
+# Copyright (C) The Adversarial Robustness Toolbox (ART) Authors 2023
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+# persons to whom the Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+# Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""
+This module implements the composite adversarial attack by sequentially perturbing different components of the inputs.
+It uses order scheduling to search for the attack sequence and uses the iterative gradient sign method to optimize the
+perturbations in semantic space and Lp-ball (see `FastGradientMethod` and `BasicIterativeMethod`).
+
+| Paper link: https://arxiv.org/abs/2202.04235
+"""
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import logging
+
+from typing import Optional, Tuple, List, TYPE_CHECKING
+
+import numpy as np
+from tqdm.auto import tqdm
+
+from art.attacks.attack import EvasionAttack
+from art.config import ART_NUMPY_DTYPE
+from art.estimators.estimator import BaseEstimator, LossGradientsMixin
+from art.estimators.classification.classifier import ClassifierMixin
+from art.utils import compute_success, check_and_transform_label_format
+
+if TYPE_CHECKING:
+    # pylint: disable=C0412
+    import torch
+    from art.estimators.classification.pytorch import PyTorchClassifier
+
+logger = logging.getLogger(__name__)
+
+
+class CompositeAdversarialAttackPyTorch(EvasionAttack):
+    """
+    Implementation of the composite adversarial attack on image classifiers in PyTorch. The attack is constructed by
+    adversarially perturbing the hue component of the inputs. It uses order scheduling to search for the attack sequence
+    and uses the iterative gradient sign method to optimize the perturbations in semantic space and Lp-ball (see
+    `FastGradientMethod` and `BasicIterativeMethod`).
+
+    Note that this attack is intended for only PyTorch image classifiers with RGB images in the range [0, 1] as inputs.
```

Comment:
I think we need the vertical line to guarantee a new line in the generated documentation.
```suggestion
    | Note that this attack is intended for only PyTorch image classifiers with RGB images in the range [0, 1] as inputs.
```